kobo/apps/organizations/serializers.py

```diff
@@ -498,6 +498,8 @@ def validate_status(self, value):
                         retry_after=remaining_seconds,
                     )
 
+                # Resend the invitation email
+                self.instance.send_invite_email()
```

Comment:
Do not send the email before all the validation has taken place. 
It should be sent from `update()`. Probably `value = OrganizationInviteStatusChoices.PENDING` should also moved to `update()` at the very end before the object is saved. With `value = OrganizationInviteStatusChoices.RESENT` , we can know if the email should be (re)sent or not.